### PR TITLE
Explicitly note any cast not defined in the documentation is illegal.

### PR DESCRIPTION
**Pull request by mcclure**
My attempt to address #979. Does this help?

This may cause a merge conflict with #961 .

### CODE REVIEW

**Comment by Havvy** [comment]
#961 already has a merge conflict that needs to be resolved.

I think the sentence is good to add. I think it should be after line 308 instead of after the table. I'll defer to your judgement though if you disagree with that though.

Here's what it would look like:

```md
`as` can be used to explicitly perform [coercions](../type-coercions.md), as well as the following additional casts.
Any cast that does not fit either a coercion rule or an entry in the table is a compiler error.
Here `*T` means either `*const T` or `*mut T`.
```

**Comment by mcclure** [comment]
I think your way is clearer, I switched it.

File: src/expressions/operator-expr.md [hunk 1]
```diff
@@ -305,7 +305,7 @@ fn average(values: &[f64]) -> f64 {
 }
 ```
 
-`as` can be used to explicitly perform [coercions](../type-coercions.md), as well as the following additional casts.
+`as` can be used to explicitly perform [coercions](../type-coercions.md), as well as the following additional casts. Any cast that does not fit either a coercion rule or an entry in the table is a compiler error.
```
Comment (Havvy) [inline src/expressions/operator-expr.md:308]:
```suggestion
`as` can be used to explicitly perform [coercions](../type-coercions.md), as well as the following additional casts.
Any cast that does not fit either a coercion rule or an entry in the table is a compiler error.
```

As per #950

**Review by Havvy — APPROVED** [review]

**Comment by Havvy** [comment]
💟 Thanks

**Pull request merged**